Sources/Defaults/Defaults+iCloud.swift

```diff
@@ -0,0 +1,539 @@
+#if canImport(OSLog)
+import OSLog
+#endif
+#if !os(macOS)
+import UIKit
+#else
+import AppKit
+#endif
+import Combine
+import Foundation
+
+/**
+Represent different data sources available for synchronization.
+*/
+public enum DataSource {
+	/**
+	Using `key.suite` as data source.
+	*/
+	case local
+
+	/**
+	Using `NSUbiquitousKeyValueStore` as data source.
+	*/
+	case remote
+}
+
+private enum SyncStatus {
+	case idle
+	case syncing
+	case completed
+}
+
+extension Defaults {
+	/**
+	Manages `Defaults.Keys` between the locale and remote storage.
+
+	Depending on the storage, `Defaults.Keys` will be represented in different forms due to storage limitations of the remote storage. The remote storage imposes a limitation of 1024 keys. Therefore, we combine the recorded timestamp and data into a single key. Unlike remote storage, local storage does not have this limitation. Therefore, we can create a separate key (with `defaultsSyncKey` suffix) for the timestamp record.
+	*/
+	public final class iCloudSynchronizer {
+		init(remoteStorage: KeyValueStore) {
+			self.remoteStorage = remoteStorage
+			registerNotifications()
+			remoteStorage.synchronize()
+		}
+
+		deinit {
+			removeAll()
+		}
+
+		@TaskLocal static var timestamp: Date?
+
+		private var cancellables: Set<AnyCancellable> = []
+
+		/**
+		Key for recording the synchronization between `NSUbiquitousKeyValueStore` and `UserDefaults`.
+		*/
+		private let defaultsSyncKey = "__DEFAULTS__synchronizeTimestamp"
+
+		/**
+		A remote key value storage.
+		*/
+		private var remoteStorage: KeyValueStore
+
+		/**
+		A FIFO queue used to serialize synchronization on keys.
+		*/
+		private let backgroundQueue = TaskQueue(priority: .background)
```

Comment:
`.background` priority means it could potentially never sync the device is on low power mode. I think it should be `.utility` instead.